src/modules/RfqModule.sol

```diff
@@ -0,0 +1,196 @@
+
+    if (takerOrder.orderHash != keccak256(actions[0].data)) revert RFQM_InvalidTakerHash();
+
+    // Update feeds in advance, so perpPrice is up to date before we use it for the trade
+    _processManagerData(order.managerData);
+
+    // Total transfers = number of assets + 3 (cash transfer, maker fee, taker fee)
+    ISubAccounts.AssetTransfer[] memory transferBatch = new ISubAccounts.AssetTransfer[](makerOrder.trades.length + 3);
+
+    int totalCashTransfer = 0;
+
+    // Iterate over the trades in the order and sum total cash to transfer
+    for (uint i = 0; i < makerOrder.trades.length; i++) {
+      TradeData memory tradeData = makerOrder.trades[i];
+
+      if (_isPerp(tradeData.asset)) {
+        int perpDelta = _getPerpDelta(tradeData.asset, tradeData.markPrice);
+        totalCashTransfer += perpDelta.multiplyDecimal(tradeData.amount);
+      } else {
+        totalCashTransfer += tradeData.markPrice.multiplyDecimal(tradeData.amount);
+      }
+
+      transferBatch[i] = ISubAccounts.AssetTransfer({
+        asset: IAsset(tradeData.asset),
+        subId: tradeData.subId,
+        amount: tradeData.amount,
+        fromAcc: order.takerAccount,
+        toAcc: order.makerAccount,
+        assetData: bytes32(0)
+      });
+    }
+
+    // Transfer the total payment for the order
+    transferBatch[transferBatch.length - 3] = ISubAccounts.AssetTransfer({
+      asset: quoteAsset,
+      subId: 0,
+      amount: totalCashTransfer,
+      fromAcc: order.makerAccount,
+      toAcc: order.takerAccount,
+      assetData: bytes32(0)
+    });
+
+    // Transfer the fee from the maker
+    transferBatch[transferBatch.length - 2] = ISubAccounts.AssetTransfer({
+      asset: quoteAsset,
+      subId: 0,
+      amount: order.makerFee.toInt256(),
+      fromAcc: order.makerAccount,
+      toAcc: feeRecipient,
+      assetData: bytes32(0)
+    });
+
+    // Transfer the fee from the taker
+    transferBatch[transferBatch.length - 1] = ISubAccounts.AssetTransfer({
+      asset: quoteAsset,
+      subId: 0,
+      amount: order.takerFee.toInt256(),
+      fromAcc: order.takerAccount,
+      toAcc: feeRecipient,
+      assetData: bytes32(0)
+    });
+
+    // TODO: events!!!!
+
+    // Execute all trades, no need to resubmit manager data
+    subAccounts.submitTransfers(transferBatch, "");
+
+    // Return SubAccounts
+    _returnAccounts(actions, newAccIds);
+
+    return (newAccIds, newAccOwners);
+  }
+
+  /**
+   * @dev Send data to IDataReceiver contracts. Can be used to update oracles before pairing trades
+   */
+  function _processManagerData(bytes memory managerData) internal {
+    if (managerData.length == 0) return;
+    IBaseManager.ManagerData[] memory managerDatas = abi.decode(managerData, (IBaseManager.ManagerData[]));
+    for (uint i; i < managerDatas.length; i++) {
+      IDataReceiver(managerDatas[i].receiver).acceptData(managerDatas[i].data);
+    }
+  }
+
+  function _isPerp(address baseAsset) internal view returns (bool) {
```

Comment:
i don't think using an internal function only returning the map value is worth it, but not too fuzzed about it either 🤣